sources/core/Stride.Core/Storage/ObjectIdBuilder.cs

```diff
@@ -324,8 +330,73 @@ public void Write(byte* buffer, int length)
                 }
             }
         }
-        
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+
+        /// <summary>
+        /// Writes a buffer of byte to this builder.
+        /// </summary>
+        /// <param name="buffer">The buffer.</param>
+        /// <param name="count">The count.</param>
+        /// <exception cref="System.ArgumentNullException">buffer</exception>
+        /// <exception cref="System.ArgumentOutOfRangeException">count;Offset + Count is out of range</exception>
+        public void Write(ref byte buffer, int length)
+        {
+            ref var currentBlock = ref Unsafe.As<uint, byte>(ref currentBlock1);
+            var position = currentLength % 16;
+
+            currentLength += length;
+
+            // Partial block to continue?
+            if (position != 0)
+            {
+                var remainder = 16 - position;
+
+                var partialLength = length;
+                if (partialLength > remainder)
+                    partialLength = remainder;
+
+                ref var dest = ref Unsafe.Add(ref currentBlock, position);
```

Comment:
Its just the monkey porting the reference assignment although this will copy byte for byte and thats not smart.
```
var dest = currentBlock + position;
```

Reply:
If you see other archaic C++'isms, let's put a warning and punt fixing it.